source/material_model/rheology/peierls_creep.cc

```diff
@@ -334,6 +334,39 @@ namespace aspect
 
 
 
+      template <int dim>
+      std::pair<double, double>
+      PeierlsCreep<dim>::compute_approximate_log_strain_rate_and_derivative (const double log_stress,
+                                                                             const double pressure,
+                                                                             const double temperature,
+                                                                             const PeierlsCreepParameters creep_parameters) const
+      {
+        /**
+        * b = (E+P*V)/(R*T)
+        * c = std::pow(gamma, p)
+        * d = std::pow(1. - c, q)
+        * s = b*p*q*c*d/(1. - c)
+        * log_arrhenius = -b*d
+        *
+        * edot_ii = log_A + (s + n) * std::log(stress) - s * std::log(gamma*peierls_stress) + log_arrhenius
+        * deriv = (s + n)
+        */
+        const PeierlsCreepParameters p = creep_parameters;
+
+        const double b = (p.activation_energy + pressure*p.activation_volume)/(constants::gas_constant * temperature);
+        const double c = std::pow(p.fitting_parameter, p.glide_parameter_p);
+        const double d = std::pow(1. - c, p.glide_parameter_q);
+        const double s = b*p.glide_parameter_p*p.glide_parameter_q*c*d/(1. - c);
+        const double log_arrhenius = -b*d;
+
+        const double deriv = s + p.stress_exponent;
```

Comment:
This may reveal my limited knowledge about rheology, but why do we keep using the name `deriv` when there are many possible derivatives of the strain rate? As far as I understand `deriv` always stands for some sort of `stress_derivative` of the strain rate, is that correct? Except in this case it is the logarithm of the derivative? So maybe `log_stress_derivative = ...`?